python/lsst/pipe/tasks/processImage.py

```diff
@@ -132,101 +111,67 @@ def makeIdFactory(self, sensorRef):
+            detections = self.detection.run(table, inputExposure)
             sources = detections.sources
             fpSets = detections.fpSets
-            if fpSets.background:           
+            if fpSets.background:
                 backgrounds.append(fpSets.background)
 
         if self.config.doDeblend:
-            self.deblend.run(calExposure, sources, calExposure.getPsf())
+            self.deblend.run(inputExposure, sources, inputExposure.getPsf())
 
         if self.config.doMeasurement:
-            self.measurement.run(calExposure, sources, exposureId=self.getExposureId(dataRef))
-
-        if self.config.doWriteCalibrate:
-            # wait until after detection and measurement, since detection sets detected mask bits
-            # and both require a background subtracted exposure;
-            # note that this overwrites an existing calexp if doCalibrate false
-               
-            if calExposure is None:
-                self.log.warn("calibrated exposure is None; cannot save it")
-            else:
-                if self.config.persistBackgroundModel:
-                    self.writeBackgrounds(dataRef, backgrounds)
-                else:
-                    self.restoreBackgrounds(calExposure, backgrounds)
-                dataRef.put(calExposure, self.dataPrefix + "calexp")
-
-        if calib is not None:
-            self.propagateCalibFlags(calib.sources, sources)
+            self.measurement.run(inputExposure, sources, exposureId=self.getExposureId(dataRef))
 
         if sources is not None and self.config.doWriteSources:
             sourceWriteFlags = (0 if self.config.doWriteHeavyFootprintsInSources
                                 else afwTable.SOURCE_IO_NO_HEAVY_FOOTPRINTS)
             if enableWriteSources:
                 dataRef.put(sources, self.dataPrefix + 'src', flags=sourceWriteFlags)
-            
-        if self.config.doMeasurement and self.config.doWriteSourceMatches:
-            self.log.info("Matching src to reference catalogue" % (dataRef.dataId))
-            srcMatches, srcMatchMeta = self.matchSources(calExposure, sources)
 
-            normalizedSrcMatches = afwTable.packMatches(srcMatches)
-            normalizedSrcMatches.table.setMetadata(srcMatchMeta)
-            dataRef.put(normalizedSrcMatches, self.dataPrefix + "srcMatch")
-        else:
-            srcMatches = None; srcMatchMeta = None
+        srcMatches = None; srcMatchMeta = None
+        if self.config.doMeasurement and self.config.doWriteSourceMatches:
+            self.log.info("Matching src to reference catalogue")
+            try:
+                srcMatches, srcMatchMeta = self.matchSources(inputExposure, sources)
+                normalizedSrcMatches = afwTable.packMatches(srcMatches)
+                normalizedSrcMatches.table.setMetadata(srcMatchMeta)
+                dataRef.put(normalizedSrcMatches, self.dataPrefix + "srcMatch")
+            except Exception as e:
```

Comment:
Is there a more specific exception we could catch instead, or is this really a catch-anything case?